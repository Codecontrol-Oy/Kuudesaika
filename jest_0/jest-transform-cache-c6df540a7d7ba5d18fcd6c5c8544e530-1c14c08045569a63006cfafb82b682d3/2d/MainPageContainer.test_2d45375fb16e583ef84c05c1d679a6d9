'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _MainPageContainer = require('./MainPageContainer');var _MainPageContainer2 = _interopRequireDefault(_MainPageContainer);
var _sinon = require('sinon');var _sinon2 = _interopRequireDefault(_sinon);
var _reduxMockStore = require('redux-mock-store');var _reduxMockStore2 = _interopRequireDefault(_reduxMockStore);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}
var mockStore = (0, _reduxMockStore2.default)();
var dispatch = _sinon2.default.spy();
describe("<MainPageContainer />", function () {

    it("Matches snapshot", function () {
        var mainPageContainer = shallow(
        _react2.default.createElement(_MainPageContainer2.default, { dispatch: dispatch, store: mockStore({ organization: {}, case: {}, event: {} }) }));

        expect(mainPageContainer).toMatchSnapshot();
    });
});