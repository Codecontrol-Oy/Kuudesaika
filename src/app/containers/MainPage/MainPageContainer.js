import React from 'react';
import { connect } from 'react-redux';
import theme from '../../theme/global.scss';
import Section from 'grommet/components/Section';
import Heading from 'grommet/components/Heading';
import Split from 'grommet/components/Split';
import Box from 'grommet/components/Box';
import {OrganizationCardList, Loader, LatestCaseList, LatestEventList, OrganizationSearch} from 'Components';
import {fetchLatest, fetchLatestEvents, fetchOrganizations, setCity} from 'Actions';

@connect((store) => {
  return {
    organization: store.organization,
    case: store.case,
    event: store.event
  };
})

export default class MainPageContainer extends React.Component {
  constructor (props) {
    super(props);
  }

  componentDidMount = () => {
      setCity(this.props.params.city);
      fetchOrganizations();
      fetchLatest();
      fetchLatestEvents();
  }

  render () {
    return (
        <Section>
          <Split flex={"left"} showOnResponsive={"both"}>
            <Box pad="small">
              <Heading className={theme.sectionTitle} tag={"h3"} uppercase>Suosituimmat organisaatiot</Heading>
              {this.props.organization.organizations ? <OrganizationCardList organizations={this.props.organization.organizations.results} /> : <Loader />}
            </Box>
            <Box pad="small">
              <Heading className={theme.sectionTitle} tag={"h3"} uppercase>Organisaatiohaku</Heading>
              {this.props.organization.organizations ? <OrganizationSearch organizations={this.props.organization.organizations.results} /> : <Loader />}
              <Heading className={theme.sectionTitle} tag={"h3"} uppercase>Viimeisimmät kokoukset</Heading>
              {this.props.event.latest_events && this.props.organization.organizations ? <LatestEventList latestEvents={this.props.event.latest_events.results} organizations={this.props.organization.organizations.results} /> : <Loader />}
              <Heading style={{marginTop: '20px'}} className={theme.sectionTitle} tag={"h3"} uppercase>Viimeisimmät Asiat</Heading>
              {this.props.case.latest_cases ? <LatestCaseList latestCases={this.props.case.latest_cases.results} /> : <Loader />}
            </Box>
          </Split>
        </Section>
    );
  }
}
