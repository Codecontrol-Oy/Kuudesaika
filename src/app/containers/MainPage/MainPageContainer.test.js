import React from 'react';
import MainPageContainer from './MainPageContainer';
import sinon from 'sinon';
import configureStore from 'redux-mock-store';
const mockStore = configureStore();
const dispatch = sinon.spy();
describe("<MainPageContainer />", () => {

    it("Matches snapshot", () => {
        const mainPageContainer = shallow(
            <MainPageContainer dispatch={dispatch} store={mockStore({ organization: {}, case: {}, event: {} })} />
        );
        expect(mainPageContainer).toMatchSnapshot();
    });
}) 
