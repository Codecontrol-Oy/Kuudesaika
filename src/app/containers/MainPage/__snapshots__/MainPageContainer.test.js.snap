// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPageContainer /> Matches snapshot 1`] = `
<MainPageContainer
  case={Object {}}
  dispatch={[Function]}
  event={Object {}}
  organization={Object {}}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
